File: src/main/java/cn/chahuyun/session/exception/ExceptionHandle.java
package cn.chahuyun.session.exception;

import kotlin.coroutines.CoroutineContext;
import kotlinx.coroutines.CoroutineExceptionHandler;
import org.jetbrains.annotations.NotNull;

/**
 * 全局异常处理
 *
 * @author Moyuyanli
 * @date 2024/1/15 11:20
 */
public class ExceptionHandle implements CoroutineExceptionHandler {
    @Override
    public void handleException(@NotNull CoroutineContext coroutineContext, @NotNull Throwable throwable) {

    }
}

File: src/main/kotlin/cn/chahuyun/session/exception/ExceptionHandle.kt
package cn.chahuyun.session.exception

import cn.chahuyun.session.HuYanSession
import kotlinx.coroutines.CoroutineExceptionHandler
import kotlin.coroutines.CoroutineContext

/**
 * 全局异常处理
 *
 * @author Moyuyanli
 * @date 2024/1/15 11:20
 */
class ExceptionHandle : CoroutineExceptionHandler {

    override val key: CoroutineContext.Key<*>
        get() = CoroutineExceptionHandler.Key

    override fun handleException(context: CoroutineContext, exception: Throwable) {
        HuYanSession.INSTANCE.logger.error(exception)
    }
}